nucleus/admin/rest/rest-service/src/main/java/org/glassfish/admin/rest/resources/custom/LogNamesResource.java

```diff
@@ -66,35 +83,46 @@ private String getLogNames(String instanceName, String type) throws IOException
 
     }
 
-    private String convertQueryResult(List<String> files, String type) {
-        StringBuilder sb = new StringBuilder();
-        String sep = "";
-        if (type.equals("json")) {
-            sb.append("{\"InstanceLogFileNames\": [");
-        } else {
-            sb.append("<InstanceLogFileNames>\n");
+    private Response convertQueryResult(List<String> files, String type) throws Exception {
+        Object entity;
+
+        switch (type) {
+            case APPLICATION_JSON:
+                entity = new JSONObject().put("InstanceLogFileNames", files);
+                break;
+            case APPLICATION_XML:
+                Writer xml = new StringWriter();
+
+                XMLStreamWriter writer = newDefaultFactory().createXMLStreamWriter(xml);
+                writer.writeStartElement("InstanceLogFileNames");
+                for (String file : files) {
+                    writer.writeEmptyElement(file);
+                }
+                writer.writeEndElement();
+                writer.flush();
+                writer.close();
```

Comment:
It would be better to move that to finally (it is not AutoCloseable, but has the close method); flush is called in close too.